### PR TITLE
Add FXIOS-10753 - [Toolbar Redesign] - Custom homepage doesn't work

### PR DESCRIPTION
**Pull request by PARAIPAN9**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10753)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/23471)

## :bulb: Description

- Adds the possibility of having a `Custom Homepage URL` when tapping the `+` button.

### Video
https://github.com/user-attachments/assets/f55649c2-0c59-4254-b6bd-23ccdffdc675

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift [hunk 1]
```diff
@@ -4101,9 +4101,15 @@ extension BrowserViewController: TopTabsDelegate {
     }
 
     func topTabsDidPressNewTab(_ isPrivate: Bool) {
-        openBlankNewTab(focusLocationField: true, isPrivate: isPrivate)
-        overlayManager.openNewTab(url: nil,
-                                  newTabSettings: newTabSettings)
+        let shouldLoadCustomHomePage = isToolbarRefactorEnabled && NewTabAccessors.getHomePage(profile.prefs) == .homePage
```
Comment (MattLichtenstein) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:4104]: Why do we only load the custom homepage for the new toolbar and not the old one?

**Review by MattLichtenstein — APPROVED** [review]
Looks good - one question

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift (hunk 1 above)
Comment (PARAIPAN9) [reply firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:4104]: For the old one it is triggered by tapping the `home` button which we don't have for the new toolbar.

**Comment by PARAIPAN9** [comment]
@Mergifyio backport release/v134

**Pull request merged**